sdk/eventhub/event-hubs/src/partitionPump.ts

```diff
@@ -38,23 +38,38 @@ export class PartitionPump {
 
   async start(): Promise<void> {
     this._isReceiving = true;
+    let requestedDefaultPosition: EventPosition | undefined;
     try {
-      await this._partitionProcessor.initialize();
+      requestedDefaultPosition = await this._partitionProcessor.initialize();
     } catch {
       // swallow the error from the user-defined code
     }
+
+    this._initialEventPosition = getInitialPosition(
+      this._initialEventPosition,
+      requestedDefaultPosition
+    );
+
+    // this is intentionally not await'd - the _receiveEvents loop will continue to
+    // execute and can be stopped by calling .stop()
     this._receiveEvents(this._partitionProcessor.partitionId);
     log.partitionPump("Successfully started the receiver.");
   }
 
   private async _receiveEvents(partitionId: string): Promise<void> {
+    if (this._initialEventPosition == null) {
+      throw new Error(
```

Comment:
Where does this error end up? Will the user see it in their processError handler?